# A plotting R script produced by the Revigo server at http://revigo.irb.hr/
# If you found Revigo useful in your work, please cite the following reference:
# Supek F et al. "REVIGO summarizes and visualizes long lists of Gene Ontology
# terms" PLoS ONE 2011. doi:10.1371/journal.pone.0021800

# --------------------------------------------------------------------------
# If you don't have the ggplot2 package installed, uncomment the following line:
# install.packages( "ggplot2" );
library( ggplot2 );

# --------------------------------------------------------------------------
# If you don't have the scales package installed, uncomment the following line:
# install.packages( "scales" );
library( scales );

# --------------------------------------------------------------------------
# Here is your data from Revigo. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","frequency","plot_X","plot_Y","log_size","value","uniqueness","dispensability");
revigo.data <- rbind(c("GO:0000086","G2/M transition of mitotic cell cycle",0.318489132256803,6.51838221757188,-0.658611254206265,1.76342799356294,-3.12292208944017,0.982548494497339,0.00305981),
c("GO:0001736","establishment of planar polarity",0.145275744538191,6.66355496570938,1.76779238230444,1.43136376415899,-1.52072502584437,0.995784321573747,0),
c("GO:0002479","antigen processing and presentation of exogenous peptide antigen via MHC class I, TAP-dependent",0.0111750572721685,6.43208126929284,3.47180027632119,0.477121254719662,-2.20216607626768,0.988630104986148,0),
c("GO:0006091","generation of precursor metabolites and energy",2.26853662625021,-1.95916729560246,6.75245334104158,2.60959440922522,-2.38485308709314,0.918423988764918,0.1090964),
c("GO:0006101","citrate metabolic process",0.0279376431804213,-3.29258649960051,8.19387189653648,0.778151250383644,-1.68988714000045,0.926656642778538,0.34215313),
c("GO:0006302","double-strand break repair",1.1342683131251,-4.58251421814293,1.98017694144718,2.3096301674259,-5.85412823894027,0.748646039392696,0.35799967),
c("GO:0006325","chromatin organization",3.4083924680114,2.04489653661957,5.23563601126212,2.78604121024255,-2.8610427566428,0.924744283949344,0.2297948),
c("GO:0006353","DNA-templated transcription termination",0.145275744538191,-4.62412728554205,4.60581520670268,1.43136376415899,-2.82541310710134,0.846903491471675,0.29164491),
c("GO:0006354","DNA-templated transcription elongation",0.245851259987707,-5.49066351287333,4.502106758638,1.65321251377534,-3.31381085201177,0.840259994581942,0.30438911),
c("GO:0006356","regulation of transcription by RNA polymerase I",0.251438788623792,-3.6692876060403,-5.60943243887135,1.66275783168157,-1.75126017356345,0.9592543372252,0.24631394),
c("GO:0006360","transcription by RNA polymerase I",0.150863273174275,-5.17225152761255,4.91946355815925,1.44715803134222,-3.06892321667802,0.846445306549496,0.2925235),
c("GO:0006363","termination of RNA polymerase I transcription",0.0167625859082528,-5.72495982888601,5.4406683986964,0.602059991327962,-1.64339280756313,0.869234187531265,0.20735785),
c("GO:0006368","transcription elongation by RNA polymerase II promoter",0.217913616807286,-5.81223647071978,4.73151905584302,1.60205999132796,-3.22022065791644,0.83975878429358,0.30136955),
c("GO:0006402","mRNA catabolic process",0.709616136782701,-4.48779842708319,0.439775093953113,2.10720996964787,-18.625923088062,0.731199672507307,0.00334352),
c("GO:0006403","RNA localization",1.01693021176734,2.54769984718726,-5.40013768308703,2.26245108973043,-2.4903317013415,0.914075364650259,0.39814636),
c("GO:0006413","translational initiation",0.37995194725373,-6.22238478918165,2.18178205387381,1.83884909073726,-11.859495819147,0.797127986003792,0.32078452),
c("GO:0006417","regulation of translation",2.47527518578533,-3.46455763353833,-4.94408111869679,2.64738297011462,-5.21050497085116,0.928923354126385,0.21922429),
c("GO:0006457","protein folding",1.23484382857462,5.62308236514963,2.60161913571198,2.34635297445064,-1.72765640127474,0.995836285046383,0.00357257),
c("GO:0006470","protein dephosphorylation",1.00016762585908,-7.56305619594511,0.312733152416653,2.25527250510331,-1.35551325685959,0.847214158071274,0.44933797),
c("GO:0006605","protein targeting",1.38011957311281,2.47112699494145,-4.97298525195051,2.39445168082622,-10.5313162746112,0.858822790709033,0.0036224),
c("GO:0006783","heme biosynthetic process",0.145275744538191,-6.88391240168903,2.83434080069682,1.43136376415899,-1.68988714000045,0.852797488189098,0.39819237),
c("GO:0006892","post-Golgi vesicle-mediated transport",0.541990277700173,1.76333965857348,-5.87037844201704,1.99122607569249,-1.67738264792503,0.927614725852721,0.25357514),
c("GO:0006997","nucleus organization",0.815779180868302,3.41793215950808,6.16815984964435,2.16731733474818,-1.6430842867006,0.907381193916546,0.34781302),
c("GO:0007005","mitochondrion organization",2.4864502430575,2.99627961117813,5.88129063802292,2.64933485871214,-12.0591208043053,0.896904187611909,0.20960009),
c("GO:0007032","endosome organization",0.508465105883668,3.47578022062673,5.62640982205212,1.96378782734556,-1.95958255831422,0.908912730335356,0.32949412),
c("GO:0007164","establishment of tissue polarity",0.145275744538191,6.67314731179438,1.38433831872146,1.43136376415899,-1.52072502584437,0.995784321573747,0.27916568),
c("GO:0008213","protein alkylation",0.832541766776555,-7.37607208113157,-0.650806424026216,2.17609125905568,-2.38485308709314,0.876589452104375,0.37659492),
c("GO:0009141","nucleoside triphosphate metabolic process",1.16220595630553,-6.62399866217774,4.19618669865065,2.32014628611105,-3.22022065791644,0.820470569701127,0.22934763),
c("GO:0009411","response to UV",0.860479409956976,-1.29947023730356,-2.90001701934704,2.19033169817029,-1.35929042961548,0.981542949594258,0.4294407),
c("GO:0009451","RNA modification",0.988992568586914,-6.43708272739024,1.70798830117336,2.25042000230889,-2.70904724675483,0.810495225769528,0.36285216),
c("GO:0010257","NADH dehydrogenase complex assembly",0.329664189528971,1.38307511949062,6.91280671309078,1.77815125038364,-3.30135344027419,0.891602609925474,0.4581782),
c("GO:0010499","proteasomal ubiquitin-independent protein catabolic process",0.0167625859082528,-6.77872941757698,-2.44312846667037,0.602059991327962,-1.87275138923524,0.893142100530889,0.4009037),
c("GO:0010948","negative regulation of cell cycle process",1.50304520310667,-1.09254840599297,-0.847567691615813,2.43136376415899,-4.82546707270884,0.89357557386083,0.18867384),
c("GO:0016236","macroautophagy",1.12309325585294,-6.45586132949508,-3.7140704221877,2.30535136944662,-3.43220131508216,0.893054538631076,0.46168367),
c("GO:0016570","histone modification",2.17913616807286,-7.18760968987272,0.0717932593241111,2.59217675739587,-3.77026747927609,0.864296989682451,0.12644465),
c("GO:0018205","peptidyl-lysine modification",1.64273341900877,-7.07543756928557,-0.255197030168212,2.46982201597816,-2.06625588960965,0.86259895752188,0.4088501),
c("GO:0019080","viral gene expression",0.290551489076382,3.80065179618619,1.60141581191878,1.72427586960079,-11.859495819147,0.995351215693001,0),
c("GO:0022406","membrane docking",0.52522769179192,-1.1386615192525,8.58286702903247,1.97772360528885,-1.40204029757394,0.996170498617508,0.003231),
c("GO:0022613","ribonucleoprotein complex biogenesis",2.49203777169358,2.07081524037878,6.910273108872,2.65030752313194,-21.19559976933,0.896080033665787,0),
c("GO:0031055","chromatin remodeling at centromere",0.0447002290886741,4.87651413713783,4.49474973700366,0.954242509439325,-1.78931937703248,0.936040731602738,0.13574611),
c("GO:0031503","protein-containing complex localization",0.849304352684807,1.19758158093737,-5.83618769006726,2.1846914308176,-2.76907911674912,0.936866148436542,0.22440019),
c("GO:0032259","methylation",1.80477174945522,-2.07194575548805,8.49501405309148,2.51054501020661,-5.02170363102011,0.943212348747809,0.07016729),
c("GO:0032479","regulation of type I interferon production",0.575515449516679,-4.39032897182897,-5.16209367623161,2.01703333929878,-1.72765640127474,0.965158924368048,0.18757796),
c("GO:0032606","type I interferon production",0.0279376431804213,-4.47391599808364,6.99586561473792,1.43136376415899,-1.90881584650529,0.913841445609851,0.1903624),
c("GO:0033044","regulation of chromosome organization",1.40805721629323,5.22882114448807,0.699221120792894,2.40312052117582,-2.27209991398738,0.981084612334153,0.0429253),
c("GO:0033108","mitochondrial respiratory chain complex assembly",0.553165334972342,2.17766838658452,6.42461653593176,2,-4.21598920419897,0.849951090366714,0.48274742),
c("GO:0033238","regulation of amine metabolic process",0.184388444990781,-1.49500368371157,-6.52609510767458,1.53147891704226,-1.85598241409871,0.971322332674008,0.1075051),
c("GO:0034248","regulation of amide metabolic process",2.8049393753143,-2.5094271602776,-6.22566733085043,2.70156798505593,-3.87863960251278,0.962792781901151,0.20248679),
c("GO:0034470","ncRNA processing",2.41939989942448,-5.42597275141623,3.53903406525899,2.63748972951251,-16.5117142420625,0.755926451191327,0.32627675),
c("GO:0035751","regulation of lysosomal lumen pH",0.156450801810359,3.98865851310126,5.76358143065542,1.46239799789896,-2.21266482406078,0.904763633704865,0.29124901),
c("GO:0042254","ribosome biogenesis",1.68184611946136,1.94951215102179,7.14394138704589,2.48000694295715,-22.9199160050964,0.865728398959248,0.41746946),
c("GO:0043248","proteasome assembly",0.0726378722690954,1.0635326317702,7.2346310987726,1.14612803567824,-1.33562084266507,0.903794342280401,0.39885465),
c("GO:0043414","macromolecule methylation",1.54774543219534,-6.3222626937878,-0.0548177956247954,2.44404479591808,-4.40765013223197,0.809899069312045,0.39028383),
c("GO:0043543","protein acylation",1.02251774040342,-7.57534028746627,-0.326147374791145,2.26481782300954,-1.64416302821014,0.87415033460305,0.38579981),
c("GO:0043687","post-translational protein modification",0.374364418617645,-7.51363743185352,-1.06504347935961,1.83250891270624,-3.44365572526668,0.885248701699106,0.34462159),
c("GO:0045023","G0 to G1 transition",0.0167625859082528,6.54723298419401,-0.411016759893081,0.602059991327962,-1.86385227737187,0.987996187036514,0.39321627),
c("GO:0045333","cellular respiration",1.07280549812818,-0.266674772945923,2.77727576182496,2.28555730900777,-6.04233876808796,0.875872958682364,0.09572648),
c("GO:0045943","positive regulation of transcription by RNA polymerase I",0.189975973626865,-2.70930093951278,-4.87805201711664,1.54406804435028,-1.32903145982689,0.95533838874936,0.23963551),
c("GO:0050658","RNA transport",0.888417053137397,2.23325707345479,-5.20481947549937,2.20411998265592,-2.15602228355215,0.886557034982307,0.49109123),
c("GO:0051607","defense response to virus",1.40246968765715,-5.79461350432895,-4.59071047382866,2.40140054078154,-1.4986589552353,0.961642780586492,0.09464774),
c("GO:0060071","Wnt signaling pathway, planar cell polarity pathway",0.229088674079455,1.52605846520014,-0.561244282409068,1.6232492903979,-2.05187187264235,0.951480896109954,0.17568784),
c("GO:0060271","cilium assembly",1.79918422081913,2.62478332926826,6.42399811367775,2.5092025223311,-3.9153890298603,0.858117732423048,0.42112122),
c("GO:0060760","positive regulation of response to cytokine stimulus",0.357601832709393,5.37358790428754,-2.98909467816496,1.81291335664286,-1.41101338453649,0.977542053717905,0.24224711),
c("GO:0061013","regulation of mRNA catabolic process",1.03928032631167,-3.1861456543905,-5.49089327408363,2.2718416065365,-5.25751912375382,0.920758103236461,0.28677598),
c("GO:0061418","regulation of transcription from RNA polymerase II promoter in response to hypoxia",0.0614628149969269,-2.76705014513491,-3.63918288330449,1.07918124604762,-2.14836253545625,0.912792938600959,0.3723853),
c("GO:0061919","process utilizing autophagic mechanism",1.74330893445829,3.94149675385708,-0.350191406707227,2.49554433754645,-2.23373037040389,0.9956842852517,0.00373171),
c("GO:0070498","interleukin-1-mediated signaling pathway",0.111750572721685,0.726700067987721,-1.13157742256815,1.32221929473392,-1.93257710684967,0.960410713603656,0.28727259),
c("GO:0070646","protein modification by small protein removal",0.949879868134324,-6.95903864459999,-0.912190377624062,2.23299611039215,-3.62431133654107,0.860611475721208,0.38244849),
c("GO:0070972","protein localization to endoplasmic reticulum",0.374364418617645,2.86395177544747,-4.91965107768209,1.83250891270624,-7.2850004329932,0.889564147192133,0.40450406),
c("GO:0071103","DNA conformation change",0.508465105883668,3.16179650802072,5.26008786683356,1.96378782734556,-4.40379784563214,0.891931832494588,0.32949412),
c("GO:0071166","ribonucleoprotein complex localization",0.00558752863608426,3.70151147531876,-4.50210395509786,0.301029995663981,-1.61435701321618,0.938015313951111,0.23073315),
c("GO:0071826","ribonucleoprotein complex subunit organization",1.16220595630553,1.25510779487253,6.36385309897364,2.32014628611105,-6.5972412646917,0.900410042619365,0.19003337),
c("GO:0072331","signal transduction by p53 class mediator",0.497290048611499,1.20131190890171,0.701308204881075,1.95424250943932,-2.18849460779842,0.948875796603548,0.44769786),
c("GO:0090305","nucleic acid phosphodiester bond hydrolysis",1.525395317651,-5.11874463022528,3.03005998991387,2.43775056282039,-3.42924723953182,0.835165197428437,0.26896549),
c("GO:0090501","RNA phosphodiester bond hydrolysis",0.955467396770408,-6.00429604490563,3.5581463182432,2.23552844690755,-2.85052000935205,0.819648431364007,0.33786959),
c("GO:0097193","intrinsic apoptotic signaling pathway",0.899592110409566,0.816595827705332,0.542803239454079,2.20951501454263,-2.40607345701386,0.940136343520193,0.04069283),
c("GO:0098781","ncRNA transcription",0.240263731351623,-4.98012036870528,4.3381177998397,1.64345267648619,-5.09660372502616,0.840562287045858,0.25764375),
c("GO:0099111","microtubule-based transport",1.14544337039727,1.57359446468836,-5.32510230450778,2.31386722036915,-4.45284842444463,0.926142954967768,0.27525497),
c("GO:0140053","mitochondrial gene expression",0.787841537687881,-4.61966661090795,5.89675548338554,2.15228834438306,-9.9667509836065,0.887054828146358,0.11195647),
c("GO:1901796","regulation of signal transduction by p53 class mediator",0.575515449516679,5.63086822719716,-2.37857533257094,2.01703333929878,-2.14836253545625,0.976149874086422,0.32988866),
c("GO:1903311","regulation of mRNA metabolic process",1.7041962340057,-2.65745578379094,-5.57683567479449,2.48572142648158,-7.18614842297618,0.954909843282931,0.15230965),
c("GO:1904893","negative regulation of receptor signaling pathway via STAT",0.139688215902106,4.5217140886656,-2.16989538853305,1.41497334797082,-1.87275138923524,0.95797686081793,0.28530787),
c("GO:1990542","mitochondrial transmembrane transport",0.497290048611499,1.59345764943218,-4.71808002612235,1.95424250943932,-1.72936046986113,0.92708219675397,0.25129842),
c("GO:2001233","regulation of apoptotic signaling pathway",2.01709783762642,5.32839915746689,-2.44902856366306,2.55870857053317,-2.44043842405411,0.972946278185862,0.0315344));

one.data <- data.frame(revigo.data);
names(one.data) <- revigo.names;
one.data <- one.data [(one.data$plot_X != "null" & one.data$plot_Y != "null"), ];
one.data$plot_X <- as.numeric( as.character(one.data$plot_X) );
one.data$plot_Y <- as.numeric( as.character(one.data$plot_Y) );
one.data$log_size <- as.numeric( as.character(one.data$log_size) );
one.data$value <- as.numeric( as.character(one.data$value) );
one.data$frequency <- as.numeric( as.character(one.data$frequency) );
one.data$uniqueness <- as.numeric( as.character(one.data$uniqueness) );
one.data$dispensability <- as.numeric( as.character(one.data$dispensability) );
#head(one.data);

chosen <- read_excel("/scratch/trcanmed/DE_RNASeq/local/share/data/chosen_for_all_revigo.xlsx", 
                     sheet = "LMO_LMX_down")
chosen$chosen <- "YES"
#chosen <- rbind(chosen , c("muscle organ development", "YES"))

one.data <- merge(one.data, chosen, by = "description", all.x = TRUE)
one.data$chosen[is.na(one.data$chosen)] <- "NO"
one.data <- one.data[, c(2,1,3,4,5,6,7,8,9,10)]
rownames(one.data) <- one.data$term_ID
one.data$new_log_size <- NA

for (i in rownames(one.data)) {
  if (one.data[i, "log_size"] < 1.62) {
    one.data[i, "new_log_size"] <- 3
  } else if (one.data[i, "log_size"] > 1.62 & one.data[i, "log_size"] < 2.2 | one.data[i, "log_size"] == 1.62) {
    one.data[i, "new_log_size"] <- 5
  } else {
    one.data[i, "new_log_size"] <- 8
  }
}

# --------------------------------------------------------------------------
# Names of the axes, sizes of the numbers and letters, names of the columns,
# etc. can be changed below
textSize <- 5
largerSize <- textSize + 2


unmute_theme <- theme_bw() +
  theme(
    text = element_text(size = textSize, family='sans'),
    axis.title = element_text(size = largerSize),
    axis.text.x = element_text(size = textSize, color="black"),#, angle = 90, vjust = 0.5, hjust=1)
    axis.text.y = element_text(size = textSize, color="black"),
    plot.title = element_text(size = largerSize, hjust = 0.5),
    legend.title = element_text(size=largerSize),
    legend.text = element_text(size=textSize)
  )


p1 <- ggplot( data = one.data );
p1 <- p1 + geom_point(aes( plot_X, plot_Y, colour = value, size = new_log_size))# + scale_size_area();
p1 <- p1 + scale_colour_gradient(low = "red", high = "yellow")#, limits = c( min(one.data$value), 5) );
p1 <- p1 + geom_point( aes(plot_X, plot_Y, size = new_log_size), shape = 21, fill = "transparent", colour = I (alpha ("white", 0.6) ))# + scale_size_area();
#p1 <- p1 + scale_size(one.data$new_log_size) + theme_bw(); # + scale_fill_gradientn(colours = heat_hcl(7), limits = c(-300, 0) );
p1 <- p1 + scale_size_continuous(labels = c(1.62, 2.13, 3.0), breaks = c(3, 5, 8)) + theme_bw()
ex <- one.data [ one.data$chosen == "YES", ];
#ex <- one.data [ one.data$dispensability < 0.15, ]
p1 <- p1 + geom_label_repel(data = ex, aes(plot_X, plot_Y, label = description), color = 'black', size =  (6/2.8),
                            box.padding = unit(0.35, "lines"),
                            point.padding = unit(0.5, "lines"),
                            segment.color = 'grey50'
)+unmute_theme;
p1 <- p1 + labs (y = "semantic space x", x = "semantic space y", size = "log_size");
p1 <- p1 + theme(legend.key = element_blank()) ;
one.x_range = max(one.data$plot_X) - min(one.data$plot_X);
one.y_range = max(one.data$plot_Y) - min(one.data$plot_Y);
p1 <- p1 + xlim(min(one.data$plot_X)-one.x_range/10,max(one.data$plot_X)+one.x_range/10);
p1 <- p1 + ylim(min(one.data$plot_Y)-one.y_range/10,max(one.data$plot_Y)+one.y_range/10);
p1 <- p1 + unmute_theme
ggsave("revigo_lmo_lmx_down.pdf", width = 120, height = 107, useDingbats=FALSE, units = "mm")
p1
dev.off()

